app/src/main/java/com/example/jrdeveloper/MainActivity.kt

```diff
@@ -2,46 +2,129 @@ package com.example.jrdeveloper
 
 import android.os.Bundle
 import androidx.activity.ComponentActivity
-import androidx.activity.compose.setContent
-import androidx.activity.enableEdgeToEdge
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.padding
-import androidx.compose.material3.Scaffold
-import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
-import com.example.jrdeveloper.ui.theme.JrDeveloperTheme
+import android.annotation.SuppressLint
+import android.os.PersistableBundle
+import android.util.Log
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
 
 class MainActivity : ComponentActivity() {
+
+    private lateinit var bank: BankAccount
+    private val bankObserver = BankLifecycleObserver()
+    private val lifecycleState: TextView = findViewById(R.id.LifecycleState)
```

Comment:
findViewById relies on the view hierarchy, which is only available after setContentView is called in onCreate. Initializing lifecycleState at the class level tries to access the view too early, causing a crash. To fix this, declare lifecycleState as lateinit and initialize it after setContentView in onCreate.

Reply:
Does this look correct?
I changed: 
private val lifecycleState: TextView = findViewById(R.id.LifecycleState) 
to 
private lateinit var lifecycleState: TextView

Under the setContentView(R.layout.activity_banking):
        setContentView(R.layout.activity_banking)
        lifecycleState = findViewById(R.id.LifecycleState)
        Log.d(bankObserver.lifecycle_tag, "OnCreate called")
        lifecycleState.text = "Activity is being created (onCreate)"